src/lib.rs

```diff
@@ -118,11 +118,42 @@ extern "C" {
         bits_stored: u16,
         high_bit: u16,
         pixel_representation: u16,
-    ) -> pixel_data;
+    ) -> PixelDataInternal;
 
     fn c_free_buffer(buffer_ptr: *const c_uchar);
 }
 
+/// Wraps the returned pixeldata
+///
+/// Exposes the data via a reference and cleans up
+/// allocated data on c side on drop of the struct
+pub struct PixelData {
+    size: usize,
+    data: NonNull<u8>,
+}
+
+impl Drop for PixelData {
+    fn drop(&mut self) {
+        unsafe {
+            c_free_buffer(self.data.as_ptr());
+        }
+    }
+}
+
+impl PixelData {
+    pub fn new(data: *const u8, size: usize) -> Self {
```

Comment:
This function can break the expected invariants for the other methods relying on `unsafe`. Better make it private, and maybe even `unsafe`.
```suggestion
    unsafe fn new(data: *const u8, size: usize) -> Self {
```

Reply:
[45c3ce1](https://github.com/pevers/gdcm-rs/pull/7/commits/45c3ce11e885236e00388a6775de9340f06cc504)